docs/security/security.adoc

```diff
@@ -0,0 +1,90 @@
+--
+
+After node module installation via NPM use the Cordova CLI to add `cordova-plugin-aerogear-security`
+
+[source,javascript]
+--
+cordova plugin add cordova-plugin-aerogear-security
+--
+
+This will install the packages needed with their native dependencies from the NPM registry and properly link all native dependencies to the project's cordova Platforms.
+
+== Usage
+
+To begin using the self defence checks in your app, you'll need to import the following `SecurityService`, `SecurityCheckType` and `SecurityCheckResult` from `@aerogear/security`:
+
+[source,javascript]
+--
+import { SecurityService, SecurityCheckType, SecurityCheckResult } from '@aerogear/security'; 
+--
+
+You will also need to retrieve an instance of `SecurityService`:
+
+[source,javascript]
+--
+const securityService = new SecurityService();
+--
+
+=== Provided Self Defence Checks
+
+The following are the self defence checks provided by the SDK.
+
+*Important* A successful `check` means that the environment the application is running in is more secure than otherwise, as opposed to signalling if a certain feature is enabled.
+
+For example, a check for whether the device is Rooted should return `true` when it is *not* rooted, since this would be the more secure condition.
+
+==== Emulator Detection Check (notEmulated)
+
+To detect if the app is being run on an emulator the `notEmulated` function can be used. If the device is *not* running in an emulator then the check will pass.
+
+==== Root Access Detection Check (notRooted)
+
+To help prevent your app running in a device that has been rooted/jailbroken the `notRooted` function can be used. If the device is *not* rooted/jailbroken then the check will pass.
+
+==== Debug Mode Check (notDebugMode)
+
+To detect if the app is runing in debug mode the `notDebugMode` function can be used. If the device is *not* running in debug mode then the check will pass.
+
+==== Device Lock Check (hasDeviceLock)
+
+To detect if a device has a lock screen set the `hasDeviceLock` function can be used. If the device *has* a screen lock enabled then this check will pass
+
+
+=== Invoking a Single Self Defence Check
+
+The `SecurityService.check` function is used to invoke a check which returns a promise of type `SecurityCheckResult`. The `check` function takes a paramater of type `SecurityCheckType` which will specify the type of check to be invoked:
+
+[source,javascript]
+--
+securityService.check(SecurityCheckType.notEmulated).then((result) => {
+    // -> add your logic to handle the result
+})
+--
+
+=== Invoking Multiple Self Defence Checks
+
+To chain multiple checks together the `SecurityService.checkMany` function is used. This returns an array of `SecurityCheckResult` objects in the order they`re provided in:
+
+[source,javascript]
+--
+const checkResults = await securityService.checkMany(SecurityCheckType.notRooted, SecurityCheckType.notEmulated);
+--
+
+=== Reporting Self Defence Checks Results Via the Metrics Service
```

Comment:
It will happen automatically right ?

Reply:
We currently have a ticket for this that Aiden is working on, I just added in the heading as not to forget to add any documentation he wishes to add after he finishes the ticket.